crates/re_space_view_tensor/src/dimension_mapping.rs

```diff
@@ -1,115 +1,150 @@
-use re_types::datatypes::TensorDimension;
-
-#[derive(Clone, Debug, PartialEq, Eq, Hash, serde::Deserialize, serde::Serialize)]
-pub struct DimensionSelector {
-    pub visible: bool,
-    pub dim_idx: usize,
-}
+use egui::NumExt as _;
+use re_types::{
+    blueprint::{archetypes::TensorSliceSelection, components::TensorDimensionIndexSlider},
+    components::{TensorDimensionIndexSelection, TensorHeightDimension, TensorWidthDimension},
+    datatypes::{TensorDimension, TensorDimensionSelection},
+};
+use re_viewport_blueprint::ViewProperty;
+
+/// Loads slice selection from blueprint and makes modifications (without writing back) such that it is valid
+/// for the given tensor shape.
+///
+/// This is a best effort function and will insert fallbacks as needed.
+/// Note that fallbacks are defined on the spot here and don't use the component fallback system.
+/// We don't need the fallback system here since we're also not using generic ui either.
+///
+/// General rules for scrubbing the input data:
+/// * out of bounds dimensions and indices are clamped to valid
+/// * missing width/height is filled in if there's at least 2 dimensions.
+pub fn load_tensor_slice_selection_and_make_valid(
```

Comment:
This could really use a unit-test or two.

For instance, what happens if your dimensions are named `height, width` and the input is `height=1, width=null`? From just reading the code, it looks like maybe the output will be `height=1, width=1`.